kernel/yosys.cc

```diff
@@ -651,6 +651,8 @@ void rewrite_filename(std::string &filename)
 		filename = filename.substr(1, GetSize(filename)-2);
 	if (filename.substr(0, 2) == "+/")
 		filename = proc_share_dirname() + filename.substr(2);
+	if (filename.substr(0, 1) == "~")
+		filename = filename.replace(0, 1, getenv("HOME"));
```

Comment:
The convention is that `~/` expands to `$HOME` and `~<user>/` expands to the home directory of the given user. Either you support only the first case, then the match should be for `~/`, not just `~`, or you support the latter case as well but then you'd have to actoally parse whatever is between the `~` and the first `/` and look up that users home directory (using `getpwnam()` or `getpwnam_r()`).

Reply:
Also it might be worth wrapping it around `#ifdef _WIN32`

Reply:
@cliffordwolf Let's start with the first case... and then maybe someone else who cares about the second case can do it.

@eddiehung Good idea (but I think you mean ifndef).

I'll fix it up a bit later.